### DISCUSSION TITLE
How to find the Child Detach Reason

### DISCUSSION THREAD

**Discussion by eric-bj-kang** [openthread/openthread#7921] [Q&A] [answered]
Hi Team,

I'm using Openthread v1.1.
In the some cases, Child was Detached, when the Channel Access Failure was occurred in Parent Router or Child.

[Q.1] Would you let me know why/when this phenomenon is occurred?
[Q.2] Would you let me know how to find the reason in the source?
- Please tell me the check point to find the root cause?

Actually Child #1 was Detached.
![image](https://user-images.githubusercontent.com/108657432/179900326-331e091b-c0e7-4c78-8e8b-db30c87879a9.png)

Best Regards,
Eric.

**Comment by jwhui** [comment, marked as answer]
> [Q.1] Would you let me know why/when this phenomenon is occurred?

A child typically detaches if it can no longer communicate with its parent. In other words, messages sent to the parent are not acknowledged and/or no response is received.

> [Q.2] Would you let me know how to find the reason in the source?

https://github.com/openthread/openthread/blob/f98f89adfe7a06ad9aa95446e2f6891bd8a874da/src/core/thread/mesh_forwarder.cpp#L1122-L1143

**Comment by eric-bj-kang** [comment]
Hi Jonathan Hui,

I captured the middleware log, and I reviewed the contents.
The problem was occurred by Channel Access Fail in Parent Router / Child node.
I think this phenomenon is normal operation, because the Child was detached by Channel Access Failure.

[Q.3]Could you tell me why this phenomenon was occurred?
and
[Q.4] Is there any method to avoid this phenomenon?  By changing the Channel?!!!
- Actually the Router 3 is OTA Server, other nodes were operated as OTA Client.
- In the above architecture, Child 1 didn't receive any message from Router 3(OTA Server) only.
- In other words, other OTA Clients(Leader/Router 2) could receive the OTA message Router 3(OTA Server).

Best Regards,
Eric.

**Reply by jwhui** [reply]
Side note: please reply directly to prior comments to keep them in the same thread, rather than suggesting new answers each time.

> [Q.3]Could you tell me why this phenomenon was occurred?

Channel Access Failure indicates that every time it tried to transmit a packet, it either (1) the received signal strength was greater than the CCA threshold or (2) the radio was busy receiving a packet. In normal situations, it is fairly rare to have such a busy channel. If you have a spectrum analyzer, you can confirm whether or not the channel is actually busy.

Can you indicate what IEEE 802.15.4 radio hardware you are using?

> [Q.4] How to avoid this phenomenon? By changing the Channel?!!!

As mentioned above, it is possible that the channel has very high utilization already. If this is the case, using a different channel can help. However, I rarely seen this occur under normal conditions.

**Reply by eric-bj-kang** [reply]
I'm using NXP K32W061 for the Thread.
https://www.nxp.com/products/wireless/multiprotocol-mcus/k32w061-41-high-performance-secure-and-ultra-low-power-mcu-for-zigbeethread-and-bluetooth-le-5-0-with-built-in-nfc-option:K32W061_41

For your more understanding, I attached the captured log and the summarized contents for the logs.
Could you review what's the problem?
I just found that the child detach was occurred by CCA in the log.
- (attached) Log.04_Child.Detach.zip
  . https://drive.google.com/file/d/16R7TA9uZC8vslc9GnQLw1X_IFMCwRJJz/view
- (attached) [Log.Reveiw_2022.07.20.xlsx](https://github.com/openthread/openthread/files/9165017/Log.Reveiw_2022.07.20.xlsx)

Additionally you mentioned the below.
"(1) the received signal strength was greater than the CCA threshold or"
Could you tell me that in detail and how to check the values?

**Reply by jwhui** [reply]
Clear Channel Assessment (CCA) is implemented in the radio, below the OpenThread platform abstraction. I would suggest reaching out to NXP regarding this issue.

**Reply by eric-bj-kang** [reply]
The occurred error is Channel Access Failure.
Is it a CCA, Clear Channel Assessment?
In other words, Is Channel Access Failure the same with CCA?

**Reply by jwhui** [reply]
Yes, Clear Channel Assessment (CCA) is the operation used to perform Channel Access during Carrier Sense Multiple Access - Collision Avoidance (CSMA-CA). When CCA indicates that the channel is busy, that results in Channel Access Failure.

**Reply by eric-bj-kang** [reply]
Then it seems that the Child Detach by Channel Access Failure is normal operation.
But in the same network, the Channel Access Failure was occurred in the just ONE node, Child 1.
Is it possible?

And could you tell me that in detail and how to check the below values?
"(1) the received signal strength was greater than the CCA threshold or"

**Reply by jwhui** [reply]
As I mentioned in https://github.com/openthread/openthread/discussions/7921#discussioncomment-3203242, Channel Access Failure can occur normally, but I typically do not see it occur often in normal conditions. Only one device experiencing this issue seems suspicious since channel utilization is typically observed by all devices within the same radio range.

CCA operation is performed within the radio, outside of OpenThread. I would suggest reaching out to NXP for support on CCA.

**Reply by eric-bj-kang** [reply]
Thanks for your check and your opinion.
With your comments, I'll review the phenomenon again.
